pkg/client/restclient/config.go

```diff
@@ -62,6 +62,9 @@ type Config struct {
 	// TODO: demonstrate an OAuth2 compatible client.
 	BearerToken string
 
+	// Server requires plugin-specified authentication.
+	AuthPlugin string
```

Comment:
> My initial (self-serving) thought was that other plugins could handle config themselves. E.g. @jdef's DCOS plugin could read an environment variable, another plugin could have its own conf file.

I think that we should try to keep configuration in one location, so that the behavior of a kubeconfig file relatively predictable without needing to set particular env vars.  During the debugging process, I find "show me the content of file X" easier than, "what was your env when you ran that command".